pyanaconda/core/configuration/anaconda.py

```diff
@@ -403,6 +403,11 @@ def set_from_opts(self, opts):
         if opts.noverifyssl:
             self.payload._set_option("verify_ssl", not opts.noverifyssl)
 
+        # Set geolocation provider
+        # FIXME: This will be removed once the boot option becomes a boolean
+        if "geoloc" in opts and opts.geoloc:
+            self.ui._set_option("geolocation_provider", opts.geoloc)
```

Comment:
What does happen if `inst.geoloc` is set to 0?

Reply:
It fails, but in a way that masks it. The string "0" would be propagated all the way to the provider lookup dict in `provider_id_to_class`, which would fall back to the Fedora provider. But this is never reached because geolocation is never started - `check_if_geolocation_should_be_used` recognized the `0` and `StartGeolocation` is never called.

Reply:
It seems dangerous to rely on that. Could the option be set to an empty string instead?